### ISSUE TITLE
Controlling/setting anonymousId

### ISSUE THREAD

**Issue by lachenmayer** [segmentio/analytics-react-native#679]
## Summary

In our app, we generate a session id for every user, so we are always guaranteed to have an id to use as an anonymous id.

We would also like to be able to correlate events sent from our server (where we can control the `anonymousId` field and currently set it to the session id) with events sent from the app.

I am aware of the `alias` event, but from reading the docs I am not sure how we would use this exactly to implement this behavior. From what I can tell, this just sets the `userId` from a previous id to a current id. However, we would like the session id (and any other session ids for that user) to be associated with the same user.

Currently, the `anonymousId` value seems to be generated here:

https://github.com/segmentio/analytics-react-native/blob/4f60a84918e8f9a0bb3e8e5fbdb2412f23048f94/packages/core/src/storage/sovranStorage.ts#L37-L48

From a cursory reading of this code, it seems like this value should not be a constant - it unnecessarily generates a UUID on every launch. (Which in production is an unnecessary call into native code.)

It does seem possible to set the `userInfo` object like this:

```ts
const client = createClient({/*...*/})
const userInfo = client.userInfo.get()
if (userInfo.anonymousId !== sessionId) {
  client.userInfo.set({ ...userInfo, anonymousId: sessionId })
}
```

This seems like a hack though, as the other anonymous id is still generated, and I can't be sure that `createClient` doesn't already have events queued up with the auto-generated anonymous id (eg. app installed / app launched events).

## Proposed Solution

Ideally, I would like to be able to pass in either an `anonymousId`, or possibly even a `UserInfoState` object to the `createClient` function, like this:

```ts
const client = createClient({
  anonymousId: sessionId,
  writeKey,
  // ...
})
// or
const client = createClient({
  initialUserInfo: { anonymousId: sessionId },
  writeKey,
  // ...
})
```

Thanks a lot for taking a look at this, apart from this minor issue the v2 has been working really well for us so far.

**Comment by RohovDmytro** [comment]
Yes. Generating anonymous id manually could also fix #696

**Comment by alanjcharles** [comment]
@lachenmayer we have just released new functionality that allows you to create your own anonymousId natively and send it across the bridge to the JS side. We built it to account for a Branch.io initialization issue, but wanted to see if you think it would work for you. https://github.com/segmentio/analytics-react-native#native-anonymousid
